From: Ben Hutchings <ben@decadent.org.uk>
Date: Sun, 23 Feb 2014 00:03:24 +0000
Subject: [PATCH 097/223] cgxb4: Stop using ethtool SPEED_* constants
Patch-mainline: v3.15-rc1
Git-commit: e8b39015b591f359d464d026170dcf76f7ed5b8e
References: bsc#909577 FATE#317550

ethtool speed values are just numbers of megabits and there is no need
to add SPEED_40000.  To be consistent, use integer constants directly
for all speeds.

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c |   20 ++++++++++----------
 drivers/net/ethernet/chelsio/cxgb4/t4_hw.c      |    8 ++++----
 2 files changed, 14 insertions(+), 14 deletions(-)

--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
@@ -426,16 +426,16 @@ static void link_report(struct net_devic
 		const struct port_info *p = netdev_priv(dev);
 
 		switch (p->link_cfg.speed) {
-		case SPEED_10000:
+		case 10000:
 			s = "10Gbps";
 			break;
-		case SPEED_1000:
+		case 1000:
 			s = "1000Mbps";
 			break;
-		case SPEED_100:
+		case 100:
 			s = "100Mbps";
 			break;
-		case 40000: /* Need a SPEED_40000 in ethtool.h */
+		case 40000:
 			s = "40Gbps";
 			break;
 		}
@@ -2273,13 +2273,13 @@ static int get_settings(struct net_devic
 
 static unsigned int speed_to_caps(int speed)
 {
-	if (speed == SPEED_100)
+	if (speed == 100)
 		return FW_PORT_CAP_SPEED_100M;
-	if (speed == SPEED_1000)
+	if (speed == 1000)
 		return FW_PORT_CAP_SPEED_1G;
-	if (speed == SPEED_10000)
+	if (speed == 10000)
 		return FW_PORT_CAP_SPEED_10G;
-	if (speed == 40000) /* Need SPEED_40000 in ethtool.h */
+	if (speed == 40000)
 		return FW_PORT_CAP_SPEED_40G;
 	return 0;
 }
@@ -2309,8 +2309,8 @@ static int set_settings(struct net_devic
 		cap = speed_to_caps(speed);
 
 		if (!(lc->supported & cap) ||
-		    (speed == SPEED_1000) ||
-		    (speed == SPEED_10000) ||
+		    (speed == 1000) ||
+		    (speed == 10000) ||
 		    (speed == 40000))
 			return -EINVAL;
 		lc->requested_speed = cap;
--- a/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/t4_hw.c
@@ -3567,13 +3567,13 @@ int t4_handle_fw_rpl(struct adapter *ada
 		if (stat & FW_PORT_CMD_TXPAUSE)
 			fc |= PAUSE_TX;
 		if (stat & FW_PORT_CMD_LSPEED(FW_PORT_CAP_SPEED_100M))
-			speed = SPEED_100;
+			speed = 100;
 		else if (stat & FW_PORT_CMD_LSPEED(FW_PORT_CAP_SPEED_1G))
-			speed = SPEED_1000;
+			speed = 1000;
 		else if (stat & FW_PORT_CMD_LSPEED(FW_PORT_CAP_SPEED_10G))
-			speed = SPEED_10000;
+			speed = 10000;
 		else if (stat & FW_PORT_CMD_LSPEED(FW_PORT_CAP_SPEED_40G))
-			speed = 40000; /* Need SPEED_40000 in ethtool.h */
+			speed = 40000;
 
 		if (link_ok != lc->link_ok || speed != lc->speed ||
 		    fc != lc->fc) {                    /* something changed */
